src/common/utils/compare.ts

```diff
@@ -10,30 +17,226 @@ export interface RankedLocationSummary extends SummaryForCompare {
+        countySummary(locationInfo.full_fips_code),
+    };
+  } else {
+    return {
+      locationInfo: locationInfo,
+      metricsInfo: stateSummary(locationInfo.state_code),
+    };
+  }
+}
+
+function isCounty(location: Location) {
+  return location.county;
+}
+
+function isState(location: Location) {
+  return !location.county;
+}
+
+function isCountyOfState(location: Location, stateId: string) {
+  return location.state_code === stateId;
+}
+
+function isMetroCounty(location: Location) {
+  return location.full_fips_code && getCountyMsaCode(location.full_fips_code);
+}
+
+function isNonMetroCounty(location: Location) {
+  return location.full_fips_code && !getCountyMsaCode(location.full_fips_code);
+}
+
+export function getAllStates(): SummaryForCompare[] {
+  return locations.filter(isState).map(getLocationObj);
+}
+
+export function getAllCounties(): SummaryForCompare[] {
+  return locations.filter(isCounty).map(getLocationObj);
+}
+
+export function getAllCountiesOfState(stateId: string): SummaryForCompare[] {
+  return locations
+    .filter(isCounty)
+    .filter((location: Location) => isCountyOfState(location, stateId))
+    .map(getLocationObj);
+}
+
+export function getAllMetroCounties(): SummaryForCompare[] {
+  return locations
+    .filter(isCounty)
+    .filter((location: Location) => isMetroCounty(location))
+    .map(getLocationObj);
+}
+
+export function getAllNonMetroCounties(): SummaryForCompare[] {
+  return locations
+    .filter(isCounty)
+    .filter((location: Location) => isNonMetroCounty(location))
+    .map(getLocationObj);
+}
+
+export function getStateMetroCounties(stateId: string): SummaryForCompare[] {
+  return locations
+    .filter(isCounty)
+    .filter((location: Location) => isCountyOfState(location, stateId))
+    .filter((location: Location) => isMetroCounty(location))
+    .map(getLocationObj);
+}
+
+export function getStateNonMetroCounties(stateId: string): SummaryForCompare[] {
   return locations
-    .filter((location: Location) => !location.county)
-    .map((stateInfo: Location) => {
-      return {
-        locationInfo: stateInfo,
-        metricsInfo: stateSummary(stateInfo.state_code),
-      };
-    });
+    .filter(isCounty)
+    .filter((location: Location) => isCountyOfState(location, stateId))
+    .filter((location: Location) => isNonMetroCounty(location))
+    .map(getLocationObj);
 }
 
-export function getCountiesArr(stateId: string): SummaryForCompare[] {
+function getCountyObj(countyFips: string) {
   return locations
-    .filter(
-      (location: Location) =>
-        location.county && location.state_code === stateId,
-    )
-    .map((countyInfo: Location) => {
-      return {
-        locationInfo: countyInfo,
-        metricsInfo:
-          countyInfo.full_fips_code && countySummary(countyInfo.full_fips_code),
-      };
-    });
+    .filter(isCounty)
+    .filter((location: Location) => location.full_fips_code === countyFips)
+    .map(getLocationObj);
+}
```

Comment:
This returns an array right? Consider using [`find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead or just returning the first element of the result.